### PR TITLE
Detect yarn version to dynamically not add deprecated flags

### PR DESCRIPTION
**Pull request by jagonzalr**
## What did you implement:

Closes [#642](https://github.com/serverless-heaven/serverless-webpack/issues/642)

## How did you implement it:

This is an upgrade to a previous [PR of mine](https://github.com/serverless-heaven/serverless-webpack/pull/1246).

The package can now detect the `Yarn` version to dynamically use the flags `--frozen-lockfile` or `--immutable` depending if running v1.x (classic) or v.2x, v3.x (modern).

If will also dynamically avoid injecting the flag `--non-interactive` even if you don't disable it on the `serverless.yml` file.

## How can we verify it:

`npm run test`

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

**Review by vicary — COMMENTED** [review]
Allow my nitpicking. Just a rename and we're good to go!

File: lib/packagers/yarn.js [hunk 1]
```diff
@@ -118,20 +135,24 @@ class Yarn {
     return _.reduce(replacements, (__, replacement) => _.replace(__, replacement.oldRef, replacement.newRef), lockfile);
   }
 
-  static install(cwd, packagerOptions) {
+  static install(cwd, packagerOptions, version) {
     if (packagerOptions.noInstall) {
       return BbPromise.resolve();
     }
+    const isModern = Yarn.isModernVersion(version);
```
[the remark this answers is not in the document]
Comment (jagonzalr) [reply lib/packagers/yarn.js]: done

File: tests/packagers/yarn.test.js [hunk 2]
```diff
@@ -28,6 +28,48 @@ describe('yarn', () => {
     expect(yarnModule.mustCopyModules).toBe(false);
   });
 
+  describe('isModernVersion', () => {
```
[the remark this answers is not in the document]
Comment (jagonzalr) [reply tests/packagers/yarn.test.js]: done

**Review by j0k3r — COMMENTED** [review]
Looks great, you can squash :+1:

**Comment by jagonzalr** [comment]
> Looks great, you can squash 👍

Done @j0k3r @vicary

**Review by j0k3r — APPROVED** [review]

**Comment by j0k3r** [comment]
Good job @jagonzalr 👏🏼

**Pull request merged**